src/lib/y2partitioner/actions/edit_blk_device.rb

```diff
@@ -82,13 +82,56 @@ def title
         end
       end
 
+      # Extended partitions and LVM thin pools cannot be edited
+      #
+      # @note An error popup is shown when the edit action cannot be performed.
+      #
+      # @return [Boolean] true if the edit action can be performed; false otherwise.
       def run?
-        if blk_device.is?(:partition) && blk_device.type.is?(:extended)
-          Yast::Popup.Error(_("An extended partition cannot be edited"))
-          false
-        else
-          true
-        end
+        errors = []
+        errors << extended_partition_error
+        errors << lvm_thin_pool_error
+        errors = errors.compact
+
+        return true if errors.empty?
+
+        Yast::Popup.Error(errors.first)
```

Comment:
NP: I can live with it, but the implementation of this looks a little bit weird. I mean, adding several errors to an array to then do a compact and display only one... All that looks like the twisted version of:

```ruby
error = extended_partition_error || lvm_thin_pool_error
return true unless error
Yast::Popup.Error(error)
false
```